### PR TITLE
feat(utils): support `diff.printBasicPrototype`, Adding possibility to get rid of prototypes from the output

### PR DESCRIPTION
**Pull request by spamshaker**
### Description
simplifies copying output to apply result in unit test

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [x] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: docs/config/index.md [hunk 1]
```diff
@@ -2330,6 +2331,13 @@ Annotation that is output at the end of diff result if it's truncated.
 
 Color of truncate annotation, default is output with no color.
 
+#### diff.printBasicPrototype
+
+- **Type**: `boolean`
+- **Default**: `true`
+
+Allows to set pretty-format option printBasicPrototype for diff output
```
Comment (hi-ogawa) [inline docs/config/index.md:2339]: Isn't it already default `false` on Vitest? https://vitest.dev/guide/snapshot.html#_2-printbasicprototype-is-default-to-false

Comment (hi-ogawa) [reply docs/config/index.md:2339]: Aha, I just notice these two `diff.printBasicPrototype` and `snapshotFormat.printBasicPrototype` are different things. I think adding this option is okay to start with, but I wonder if we should have these two same.

Btw, can you add a test?

Comment (sheremet-va) [reply docs/config/index.md:2339]: It should definitely be the same as the default now (disabled)

Comment (spamshaker) [reply docs/config/index.md:2339]: take a look and tell me what will be the result of this evaluation?
https://github.com/vitest-dev/vitest/blob/main/packages/pretty-format/src/index.ts#L491

So I am not changing the default but adding missing possibility to set it from main config
If this should be default behavior (disabled)  the code I pointed must be changed

Comment (hi-ogawa) [reply docs/config/index.md:2339]: > take a look and tell me what will be the result of this evaluation?

Adding tests will hopefully tell you that :smiley: 

Btw, changing the default will probably break a few our internal diff test. If you like, you can just add option in this PR and we can follow up changing the default separately.

**Review by hi-ogawa — DISMISSED** [review]

**Comment by hi-ogawa** [comment]
Closing this PR as I merged your commits into https://github.com/vitest-dev/vitest/pull/6740 and fixed conflicts early. Thanks for the contribution!

**Pull request closed without merging**